File: JUNGOL/Beginner/#1314 문자사각형2/문자사각형2.py
n = int(input())
square = [[0 for i in range(n)] for i in range(n)]

dir,c = 1,ord('A')
for i in range(n):
    if dir == 1:
        for j in range(n):
            square[j][i] = c
            c += 1
            if c > ord('Z'): c = ord('A')
    else:
        for j in range(n-1, -1, -1):
            square[j][i] = c
            c += 1
            if c > ord('Z'): c = ord('A')

    dir = -dir

for s in square:
    for si in s:
        print(chr(si), end=' ')
    print()

File: JUNGOL/Beginner/#1314 문자사각형2/문자사각형2.cpp
#include <iostream>
#include <vector>
using namespace std;

int main(){
    int n; cin>>n;
    vector<vector<char>> square;
    for(int i=0; i<n; i++){
        vector<char> v(n);
        square.push_back(v);
    }

    int dir = 1;
    char c = 'A';
    for(int i=0; i<n; i++){
        if(dir == 1){
            for(int j=0; j<n; j++){
                square[j][i] = c++;
                if(c > 'Z') c = 'A';
            }
        }
        else{
            for(int j=n-1; j>=0; j--){
                square[j][i] = c++;
                if(c > 'Z') c = 'A';
            }
        }

        dir = -dir;
    }

    for(vector<char> s : square){
        for(char si : s) printf("%c ", si);
        cout<<endl;
    }
}
